docs/hugo/content/contributing/add-a-new-code-generated-resource/before-you-begin.md

```diff
@@ -0,0 +1,98 @@
+
+We work out these by reviewing the [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs/tree/main/specification) repo. We will walk through an example of adding Azure Synapse Workspace.
+
+### Kind
+
+Also known as the _name_ of the resource, you usually know this going in. In our example above, the name of the resource is `workspace`. 
+
+If you're not sure, one approach is to look in the Swagger/OpenAPI specification file for the service and find the documented PUT for the resource you're interested in. The resource name will be the
+[second to last section of the URL](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json#L71).
+
+Another approach is to search for the ARM documentation of the resource. Including the search terms `Azure` `ARM` and `Bicep` will help you find the right documentation. For example, we can search for [`Azure ARM Bicep Synapse Workspaces`](https://www.google.com/search?q=azure+arm+bicep+synapse+workspaces&oq=azure+arm+bicep+synapse+workspaces) and (at least at the time of writing) the first result is the [Azure Synapse Workspace](https://learn.microsoft.com/en-us/azure/templates/microsoft.synapse/workspaces?pivots=deployment-language-arm-template) resource.
+
+**Note**: ARM almost exclusively uses plural names for resources. ASO uses singular forms, aligning with the usual practices in Kubernetes. For example, the Azure Synapse Workspace resource is called `workspaces` in ARM and `workspace` in ASO.
+
+### Group
+
+This is named after the Azure service, for example `resources` or `documentdb`.
+
+In our example, this is `synapse` (from `Microsoft.Synapse`, the provider documented in the resource URL).
+
+### Version
+
+The version is the `api-version` of the resource. This is usually a date, sometimes with a `-preview` suffix. 
+
+In our example entry from above, this is `2021-06-01`.
+
+**Note**: In most cases there will be multiple API versions for a given resource, you can see this in the
+[Synapse](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/synapse/resource-manager/Microsoft.Synapse/stable)
+folder for example, there are 4 API versions as of April 2023. It is _strongly_ recommended that you use the latest
+available non-preview `api-version` when choosing the version of the resource to add.
+
+## Development Environment
+
+With the resource identified, you should ensure your [development environment]({{< relref "developer-setup" >}}) is set up to add the resource.
+
+While it _**is**_ possible to do this using just `go`, you end up doing a lot of manual work that we've automated using `task` and other tools.
+
+The recommended approaches are 
+
+* Use our .devcontainer, either on [Linux]({{< relref "developer-setup" >}}#dev-container-with-vs-code-on-linux) or [using WSL on Windows]({{< relref "developer-setup" >}}#dev-container-with-vs-code-on-windows)
```

Comment:
What about Mac?

Reply:
I think the container itself should be fine, because they build it themselves on their arch so it should be ARM64.

The issue (and I do also recall Harsh having one) would have to do with any upstream binaries we directly get that don't support ARM. I'm not sure what all binaries don't support ARM at this point though - the Kubernetes ones via envtest all should AFAIK, and ASO when built locally will as well.